### ISSUE TITLE
Default to using a `sample` dimension instead of `chain` and `draw`

### ISSUE THREAD

**Issue by OriolAbril** [arviz-devs/arviz#1951]
## Tell us about it
The information about the chain is only necessary and useful for some diagnostics, whereas for users and some others it is not necessary and we end up stacking the chain and draw dims into a sample one. It might be worth it to change this around. Default to a stacked sample dimension and unstack when necessary.

## Thoughts on implementation
* loo, loo_pit, plot_bpv... would no longer need to stack before their computations
* most plots would not have any difference. the xarray_sel_iter and similar helpers would need an option to unstack for traceplot or autocorrplot
* ess, rhat would need to unstack before their computations (like loo or loo_pit do now)

**Comment by ahartikainen** [comment]
Even when we stack, do we still need to unstack or can we still use the coords (chain, draw info)?

    idata.stack({"sample": ["chain", "draw"]})

We would need to unstack when saving the output, but the dim name still could be kept (or not?)

**Comment by OriolAbril** [comment]
Right, netcdf doesn't support saving stacked dimensions, we'd need to do something about it, this might be the main challenge to overcome.

I haven't yet checked how accessible is the chain and draw information while stacked, it might be an exploded 1d array of length n_samples. I think the main argument for this is more that this information is rarely useful, and so having to unstack in such cases is better than having to stack to get the `sample` dimension which is needed much more often.
